### PR TITLE
Fix problems according to test coverage

### PR DESCRIPTION
**Pull request by anton-hrytsevich**
Related-tesk: [APPLINK-25171](https://adc.luxoft.com/jira/browse/APPLINK-25171)

Please review @Abyzhynar, @Oherasym, @okozlovlux, @LuxoftAKutsan, @anosach-luxoft, @wolfylambova22. 
And domain expert: @dtrunov. 

Please review it's ASAP, because it need for release.

In this PR changes that make CRQ pass for testing.

### CODE REVIEW

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h [hunk 1]
```diff
@@ -85,6 +85,23 @@ namespace NsSmart = NsSmartDeviceLink::NsSmartObjects;
 std::string MergeInfos(const std::string& first, const std::string& second);
 
 /**
+ * @brief MergeInfos merge 2 infos in one string with info
```
Comment (AByzhynar) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @AGritsevich merge 2 infos `in`-> `into` one string with info

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 2]
```diff
@@ -41,6 +41,25 @@ namespace application_manager {
 
 namespace commands {
 
+std::string MergeInfos(const ResponseInfo& first_info,
+                       const std::string& first_str,
+                       const ResponseInfo& second_info,
+                       const std::string& second_str) {
+  if ((first_info.interface_state == HmiInterfaces::STATE_NOT_AVAILABLE) &&
+      (second_info.interface_state != HmiInterfaces::STATE_NOT_AVAILABLE) &&
+      false == second_str.empty()) {
```
Comment (AByzhynar) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @AGritsevich `false == second_str.empty()` -> `!second_str.empty()`

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 3]
```diff
@@ -41,6 +41,25 @@ namespace application_manager {
...
+    return second_str;
+  }
+
+  if ((second_info.interface_state == HmiInterfaces::STATE_NOT_AVAILABLE) &&
+      (first_info.interface_state != HmiInterfaces::STATE_NOT_AVAILABLE) &&
+      false == first_str.empty()) {
```
Comment (AByzhynar) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @AGritsevich The same as above

**Review by AByzhynar — CHANGES_REQUESTED** [review]

**Review by anosach-luxoft — APPROVED** [review]

**Review by OHerasym — APPROVED** [review]

**Review by dtrunov — APPROVED** [review]
Reviewed

**Review by kozlov-oleksandr — APPROVED** [review]
@AGritsevich , reviewed

**Review by wolfylambova22 — APPROVED** [review]
@AGritsevich Reviewed.

**Comment by AByzhynar** [comment]
@AGritsevich Reviewed

File: src/components/application_manager/src/commands/command_request_impl.cc (hunk 2 above)
Comment (anton-hrytsevich) [reply src/components/application_manager/src/commands/command_request_impl.cc]: @AByzhynar see  843a994

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h (hunk 1 above)
Comment (anton-hrytsevich) [reply src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @AByzhynar see  843a994

File: src/components/application_manager/src/commands/command_request_impl.cc (hunk 3 above)
Comment (anton-hrytsevich) [reply src/components/application_manager/src/commands/command_request_impl.cc]: @AByzhynar see  843a994

**Pull request merged**